test/functional/ui/screen_multigrid_spec.lua

```diff
@@ -72,6 +86,166 @@ describe('multigrid screen', function()
+      {1:~                         }|
+      {1:~                         }|
+    ## grid 3
+                                |
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+    ## grid 4
+      ^                          |
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+      {1:~                         }|
+    ]], nil, nil, function()
+      iswin(screen.win_position[2].win)
+      eq(7, screen.win_position[2].startrow)
+      eq(27, screen.win_position[2].startcol)
+      eq(26, screen.win_position[2].width)
+      eq(5, screen.win_position[2].height)
+
+      iswin(screen.win_position[3].win)
+      eq(0, screen.win_position[3].startrow)
+      eq(0, screen.win_position[3].startcol)
+      eq(26, screen.win_position[3].width)
+      eq(12, screen.win_position[3].height)
+
+      iswin(screen.win_position[4].win)
+      eq(0, screen.win_position[4].startrow)
+      eq(27, screen.win_position[4].startcol)
+      eq(26, screen.win_position[4].width)
+      eq(6, screen.win_position[4].height)
+    end)
+    command('wincmd h')
+    command('q')
+    screen:expect([[
+    ## grid 1
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+      {11:[No Name]                                            }|
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+                                {12:│}                          |
+      {12:[No Name]                                            }|
+                                                           |
+    ## grid 2
+                                                           |
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+      {1:~                                                    }|
+    ]], nil, nil, function()
+      -- TODO(utkarshme): We have resized both the grids. We should receive
```

Comment:
@bfredl This is a bug, right? Or is it the expected behavior?

Reply:
Isn't grid 4 the closed window? So the screen tests basically asserts that grid 4 has been destroyed, as expected? I guess what we also could do is to add the `win_close` event already to `ext_multigrid`. Then screen.lua could handle it simply by setting `win_position[id]` to `nil`, which this test then can check.

Reply:
There are 3 window grids just before the `close` event. Then I switch to grid 3 (`wincmd h`) and close it. Shouldn't there be 2 grids left - 2 and 4?

> I guess what we also could do is to add the `win_close` event already to `ext_multigrid`

Right. Let me do that.

Reply:
ah, I see, the problem might be that screen.lua tries to treat the table with grids as an array (which means with `{1,2,nil,4}`, the 4 will be missed) , I'll make a patch...

Reply:
@UtkarshMe If you in the function `Screen:render` change
```
  for igrid,grid in ipairs(self._grids) do
```
to
```
  for igrid,grid in pairs(self._grids) do
```
does it help?

Reply:
That did it! Yay! :)